### PR TITLE
Fix update replication cluster but not update replicator.

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

When updating replication-cluster for a topic from CLI, the broker doesn't update the replicator.

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 1]
```diff
@@ -3063,6 +3046,27 @@ public void onUpdate(TopicPolicies policies) {
         });
     }
 
+    private CompletableFuture<Void> checkReplicator() {
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:3049]: There is already a `checkReplication` in this class. It seems contains most of the logic here.

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:3049]: Yes, thanks.

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by Jason918** [comment]
@AnonHxy Please help review this.

**Review by yuruguo — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 2]
```diff
@@ -611,28 +610,13 @@ private boolean hasRemoteProducers() {
 
     public CompletableFuture<Void> startReplProducers() {
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:611]: This method can return void

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:611]: Yes, just keep the original signature.

Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:611]: +1

**Review by codelipenghui — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java (hunk 2 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:611]: There are some methods like `checkReplication`,`addProducer` that invoke this method and use `thanAccept`.
So keep the original signature is much more better.
@AnonHxy @codelipenghui

**Review by AnonHxy — APPROVED** [review]

**Review by lhotari — CHANGES_REQUESTED** [review]
It seems that this PR would revert #232 changes. @rdhabalia would you be able to review this PR to ensure that it doesn't break things?

**Comment by Technoboy-** [comment]
> It seems that this PR would revert #232 changes. @rdhabalia would you be able to review this PR to ensure that it doesn't break things?

BTW, this pr relies on https://github.com/apache/pulsar/pull/14605

**Comment by Technoboy-** [comment]
> It seems that this PR would revert #232 changes. @rdhabalia would you be able to review this PR to ensure that it doesn't break things?

I think #232 has added the test to coverage this.

**Comment by Technoboy-** [comment]
Hi, @lhotari , could you help review this again ?

**Comment by lhotari** [comment]
> Hi, @lhotari , could you help review this again ?

I'd rather have @rdhabalia take a look since he knows this area well. 
To me it looks like this PR is removing changes made in #232. It would be good to find an explanation of why that code was there in the first place and why it's now safe to remove.

**Comment by Technoboy-** [comment]
> > Hi, @lhotari , could you help review this again ?
> 
> I'd rather have @rdhabalia take a look since he knows this area well. To me it looks like this PR is removing changes made in #232. It would be good to find an explanation of why that code was there in the first place and why it's now safe to remove.

Yes, we roll back the code. But the test is there.

**Comment by Technoboy-** [comment]
Hi @merlimat , could you give some idea about this pr ?

**Comment by Technoboy-** [comment]
> > Hi, @lhotari , could you help review this again ?
> 
> I'd rather have @rdhabalia take a look since he knows this area well. To me it looks like this PR is removing changes made in #232. It would be good to find an explanation of why that code was there in the first place and why it's now safe to remove.

Hi @lhotari , I remove the modification of `startReplProducers`. #232 is too old to find the problem at that time.  So keep it as the original implementation, I will open another issue to track. 
As some users occur the above issue, we need this pr.

**Review by lhotari — APPROVED** [review]
LGTM

**Pull request merged**